### PR TITLE
feat: Change Time fields to List instead of Set

### PR DESCRIPTION
**Pull request by hbdejesus**
Closes: #40

- Change type of `bySeconds`, `byMinutes`, `byHours` to `List<int>` Since when there are times that has the same seconds, minutes or hours, after parsing they are remove since the type is `Set`

### CODE REVIEW

**Comment by hbdejesus** [comment]
@JonasWanke Can't seem to add labels. This is a Breaking Change.

File: lib/src/recurrence_rule.dart [hunk 1]
```diff
@@ -23,9 +23,9 @@ class RecurrenceRule {
     this.until,
     this.count,
     this.interval,
-    Set<int> bySeconds = const {},
-    Set<int> byMinutes = const {},
-    Set<int> byHours = const {},
+    List<int> bySeconds = const [],
+    List<int> byMinutes = const [],
+    List<int> byHours = const [],
```
Comment (JonasWanke) [inline lib/src/recurrence_rule.dart:28]: I previously converted these to a `SplayTreeSet` to ensure they have a consistent order when iterating over the elements ([`Set`](https://api.dart.dev/stable/3.0.6/dart-core/Set-class.html) doesn't guarantee this). This is no longer necessary with lists, so we can just store them directly:

```suggestion
    this.bySeconds = const [],
    this.byMinutes = const [],
    this.byHours = const [],
```

`bySeconds = List.of(bySeconds),` and Co. below can then be removed

File: lib/src/recurrence_rule.dart [hunk 2]
```diff
@@ -23,9 +23,9 @@ class RecurrenceRule {
...
     Set<ByWeekDayEntry> byWeekDays = const {},
```
Comment (JonasWanke) [inline lib/src/recurrence_rule.dart:29]: I think that it makes sense to also convert the remaining sets to lists for consistency

File: lib/src/recurrence_rule.dart [hunk 3]
```diff
@@ -95,8 +90,8 @@ class RecurrenceRule {
               [
                 // This comment is to keep the formatting of the lines below.
                 bySeconds, byMinutes, byHours,
-                byWeekDays, byMonthDays, byYearDays,
-                byWeeks, byMonths,
+                byWeekDays.toList(), byMonthDays.toList(), byYearDays.toList(),
+                byWeeks.toList(), byMonths.toList(),
```
Comment (JonasWanke) [inline lib/src/recurrence_rule.dart:94]: The `.toList()` calls can be removed when the rest are sets as well

**Review by JonasWanke — CHANGES_REQUESTED** [review]
Thanks for creating this PR! I added some comments, but, in general, the changes look good.

However, have you customized the formatter settings? The usual line limit is at 80 columns and this PR changes some code to longer lines, which my local formatter then reverts again

**Comment by hbdejesus** [comment]
> Thanks for creating this PR! I added some comments, but, in general, the changes look good.
> 
> However, have you customized the formatter settings? The usual line limit is at 80 columns and this PR changes some code to longer lines, which my local formatter then reverts again

Yeah sorry mine was set to 120. Adjusted it to 80

**Comment by hbdejesus** [comment]
@JonasWanke I'm having a hard time figuring out though what causes the `monthly_test.dart` to fail 🤔

**Comment by JonasWanke** [comment]
@hbdejesus Sorry for the delay! Code looks good now, and I just found the reason for the failing test. The following lines also have to be updated to use a list instead of a set:

https://github.com/JonasWanke/rrule/blob/a861fc93b0f3f46c1de1a781608c0074e5630790/lib/src/codecs/text/encoder.dart#L68-L71

**Comment by hbdejesus** [comment]
@JonasWanke Updated. Thanks!

**Review by JonasWanke — APPROVED** [review]

**Comment by JonasWanke** [comment]
Thanks for your contribution! I'll comment here when this change gets released, but that will still take some time as I have other breaking changes planned as well (most notably, the switch to https://github.com/JonasWanke/chrono)

**Pull request merged**